orchestra/static/orchestra/common/components/quill/js/directives.js

```diff
@@ -0,0 +1,214 @@
+          function getDropCharOffset(x, y) {
+            // Helper function for getting the x-y character offset of a drop
+            // event in a contenteditable field.
+            // Modified from http://stackoverflow.com/a/10659990.
+            var range;
+            // Standards-based way; implemented only in Firefox
+            if (document.caretPositionFromPoint) {
+                var pos = document.caretPositionFromPoint(x, y);
+                return {'offset': pos.offset, 'node': pos.offsetNode};
+            } else if (document.caretRangeFromPoint) {
+              // Webkit
+              range = document.caretRangeFromPoint(x, y);
+              return {'offset': range.startOffset, 'node': range.startContainer};
+            } else if (document.body.createTextRange) {
+              // IE doesn't natively support retrieving the character offset, so
+              // insert image at end of text
+              // TODO(jrbotros): rewrite with https://github.com/timdown/rangy
+              return;
+            }
+          }
+
+          function getAllChildNodes(parent) {
+            // Returns in-order list of all child nodes of `parent`
+            var bfsStack = [parent];
+            var nodes = [];
+            while (bfsStack.length) {
+              var currentNode = bfsStack.pop();
+              if (currentNode.childNodes) {
+                // Concat with copy of child NodeList; can't use pop operation
+                // on original
+                bfsStack = bfsStack.concat(Array.prototype.slice.call(currentNode.childNodes));
+              }
+              nodes.push(currentNode);
+            }
+            return nodes.reverse();
+          }
+
+          // http://stackoverflow.com/a/22289650
+          function getLeafNodes(parent) {
+            // Returns in-order list of all Quill leaf nodes of `parent`
+            var textNodeType = 3;
+            var nodes = getAllChildNodes(parent);
+            var leafNodes = nodes.filter(function(elem) {
+              return elem.nodeType === textNodeType || !elem.firstChild;
+            });
+            return leafNodes;
+          }
+
+          var caretPosition = getDropCharOffset(dropEvent.clientX, dropEvent.clientY);
+          if (!caretPosition) {
+            return scope.editor.getLength();
+          }
+          var leaves = getLeafNodes(editorContainer.getElementsByClassName('ql-editor')[0]);
+          var opIndex = leaves.indexOf(caretPosition.node);
+          var contents = scope.editor.getContents()
+          contents.ops = contents.ops.slice(0, opIndex)
+          return contents.length() + caretPosition.offset;
+         }
+
+        // Post image data to an API endpoint that returns an image URL, then
+        // adding it to the editor (replacing the given character range)
+        function uploadImage(file, range, e) {
+          var uploadAPIEndpoint = '/orchestra/api/interface/upload_image/'
+          var maxUploadSizeMB = 20;
```

Comment:
I went ahead and added it as an optional directive attribute with 5 MB as a fallback, let me know if that works!